build-system/server/new-server/transforms/utilities/script.ts

```diff
@@ -27,16 +28,61 @@ export interface ScriptNode extends PostHTML.Node {
   };
 }
 
+const VALID_SCRIPT_EXTENSION = ['.js', '.mjs'];
+
+function isValidScriptExtension(url: URL): boolean {
+  return VALID_SCRIPT_EXTENSION.includes(extname(url.pathname));
+}
+
 /**
  * Determines if a Node is really a ScriptNode.
  * @param node
  */
-export function isValidScript(node: PostHTML.Node): node is ScriptNode {
+export function isValidScript(node: PostHTML.Node, looseScriptSrcCheck?: boolean): node is ScriptNode {
   if (node.tag !== 'script') {
     return false;
   }
 
   const attrs = node.attrs || {};
-  const src = new URL(attrs.src || '');
-  return src.origin === VALID_CDN_ORIGIN && extname(src.pathname) === '.js';
+  const url = tryGetURL(attrs.src || '');
+  if (looseScriptSrcCheck) {
+    return isValidScriptExtension(url);
+  }
+  return url.origin === VALID_CDN_ORIGIN && isValidScriptExtension(url);
+}
+
+export function isJsonScript(node: PostHTML.Node): boolean {
+  if (node.tag !== 'script') {
+    return false;
+  }
+  const attrs = node.attrs || {};
+  const type = attrs.type || '';
+  return type.toLowerCase() === 'application/json';
+}
+
+/**
+ * Transforms a url's extension type to the desired type.
+ * ex. v0.js -> v0.mjs
+ */
+export function toExtension(url: URL, extension: string): URL {
+  const parsedPath = parse(url.pathname);
+  parsedPath.base = parsedPath.base.replace(parsedPath.ext, extension);
+  parsedPath.ext = extension;
+  url.pathname = format(parsedPath);
+  return url;
+}
+
+/**
+ * This is a temporary measure to allow for a relaxed parsing of our
+ * fixture files src url's before they are all fixed accordingly.
+ */
+export function tryGetURL(src: string, port: number = 8000): URL {
```

Comment:
I've found myself going back and forth on this rule since the URL object is all caps.

Consistency beats reasoning in this case though, so I agree and we should try to be consistent across the codebase.

Reply:
done.